src/main/java/net/dv8tion/jda/core/entities/impl/ReceivedMessage.java

```diff
@@ -0,0 +1,815 @@
+    {
+        Checks.notNull(types, "Mention Types");
+        if (types.length == 0)
+            return isMentioned(mentionable, MentionType.values());
+        for (MentionType type : types)
+        {
+            switch (type)
+            {
+                case HERE:
+                {
+                    if (isMass("@here"))
+                        return true;
+                    break;
+                }
+                case EVERYONE:
+                {
+                    if (isMass("@everyone"))
+                        return true;
+                    break;
+                }
+                case USER:
+                {
+                    if (isUserMentioned(mentionable))
+                        return true;
+                    break;
+                }
+                case ROLE:
+                {
+                    if (isRoleMentioned(mentionable))
+                        return true;
+                    break;
+                }
+                case CHANNEL:
+                {
+                    if (mentionable instanceof TextChannel)
+                    {
+                        if (getMentionedChannels().contains(mentionable))
+                            return true;
+                    }
+                    break;
+                }
+                case EMOTE:
+                {
+                    if (mentionable instanceof Emote)
+                    {
+                        if (getEmotes().contains(mentionable))
+                            return true;
+                    }
+                    break;
+                }
+//              default: continue;
+            }
+        }
+        return false;
+    }
+
+    private boolean isUserMentioned(IMentionable mentionable)
+    {
+        if (mentionable instanceof User)
+        {
+            if (getMentionedUsers().contains(mentionable))
+                return true;
+        }
+        else if (mentionable instanceof Member)
+        {
+            final Member member = (Member) mentionable;
+            if (getMentionedUsers().contains(member.getUser()))
+                return true;
+        }
+        return false;
+    }
+
+    private boolean isRoleMentioned(IMentionable mentionable)
+    {
+        if (mentionable instanceof Role)
+        {
+            if (getMentionedRoles().contains(mentionable))
+                return true;
+        }
+        else if (mentionable instanceof Member)
+        {
+            final Member member = (Member) mentionable;
+            if (CollectionUtils.containsAny(getMentionedRoles(), member.getRoles()))
+                return true;
+        }
+        else if (isFromType(ChannelType.TEXT) && mentionable instanceof User)
+        {
+            final Member member = getGuild().getMember((User) mentionable);
+            if (CollectionUtils.containsAny(getMentionedRoles(), member.getRoles()))
+                return true;
```

Comment:
This might throw a NPE if the user given is not present in the guild the message was sent in